legion/engine/physics/physx/data/physx_wrapper_container.cpp

```diff
@@ -0,0 +1,38 @@
+#include "physx_wrapper_container.hpp"
+
+namespace legion::physics
+{
+    PhysxInternalWrapper& PhysxWrapperContainer::createPhysxWrapper(physics_component& unregisteredPhysXWrapper)
+    {
+        unregisteredPhysXWrapper.physicsComponentID = nextID;
+        m_wrapperIDSet.insert(nextID);
+
+        m_physxWrappers.emplace_back();
+
+        nextID++;
+        return m_physxWrappers[m_physxWrappers.size() - 1];
+    }
+
+    void PhysxWrapperContainer::PopAndSwapRemoveWrapper(size_type id)
+    {
+        size_type index = m_wrapperIDSet.index_of(id);
+
+        if (index != m_wrapperIDSet.npos)
+        {
+            m_wrapperIDSet.erase(id);
+            std::swap(m_physxWrappers[index], m_physxWrappers[m_physxWrappers.size() - 1]);
+            m_physxWrappers.pop_back();
+        }
+    }
+
+    std::optional<std::reference_wrapper< PhysxInternalWrapper>> PhysxWrapperContainer::findWrapperWithID(size_type id)
+    {
+        if (m_wrapperIDSet.contains(id))
+        {
+            PhysxInternalWrapper& wrapper = m_physxWrappers[m_wrapperIDSet.index_of(id)];
+            return std::optional<std::reference_wrapper< PhysxInternalWrapper>>{ wrapper };
+        }
+
+        return std::nullopt;
+    }
+}
```

Comment:
wait.. isn't this exactly what sparse_map does?

Reply:
I just now looked into sparse_map and from a functionality standpoint yea pretty much. I don't really see the benefit of using an unordered_map for my specific use case tho.

Reply:
from what I can see on sparse_set.hpp , the sparse set is implemented as 2 vectors. I would argue that the sparse_set + value vector I'm using is much more cache-friendly since a std::unordered_map and std::unordered_set both use an array of linked lists (https://stackoverflow.com/questions/31112852/how-stdunordered-map-is-implemented) . Using one of these would mean that the values (in this case, physxInternalWrappers) are far apart in memory, causing more cache misses.

Reply:
ah yeah, you're right there. i meant hashed_sparse_set. the thing is that sparse_set works best with consecutive integers (1,2,3,4...) and doesn't rly work with random integers like hashes. so depending on what these id's are ig this is fine.

this is the biggest downfall of sparse sets, if you try to add the id `0x40A36E6FE89EB61C` you basically lock up the program until it has allocated memory for 4657687866554365468 integers